src/exporter/http_server.cc

```diff
@@ -134,31 +136,18 @@ void http_server(tcp::acceptor &acceptor, tcp::socket &socket) {
   });
 }
 
-std::string dns_lookup(std::string hostname) {
-  boost::asio::io_service io_service;
-  boost::asio::ip::tcp::resolver resolver(io_service);
-  boost::asio::ip::tcp::resolver::query query(hostname, "9926");
-  boost::asio::ip::tcp::resolver::iterator iter = resolver.resolve(query);
-  boost::asio::ip::tcp::endpoint endpoint = iter->endpoint();
-  std::string ip_address = endpoint.address().to_string();
-
-  return ip_address;
-}
-
-void http_server_thread_entrypoint() {
+void http_server_thread_entrypoint(std::string exporter_addrs) {
   try {
-    std::string hostname = ceph_get_short_hostname();
-
-    std::string ip_address = dns_lookup(hostname);
-    auto const address = net::ip::make_address(ip_address);
-    unsigned short port = 9926;
+    exporter_addrs = !exporter_addrs.empty() ? exporter_addrs : g_conf().get_val<std::string>("exporter_addrs");
+    auto const address = net::ip::make_address(exporter_addrs);
```

Comment:
ok..but default addrs option is set to localhost so it'll solve empty case I guess.

Reply:
Ok I understood..but I've default value `127.0.0.1` and not empty! Should tht be changed? @cbodley

Reply:
yes please! the empty default was discussed in prior review at https://github.com/ceph/ceph/pull/45220#discussion_r854525352. you'll just have to avoid calling `make_address()` when the option is empty:
```c++
net::ip::address address;
if (!exporter_addrs.empty()) {
  address = net::ip::make_address(exporter_addrs);
}
```

Reply:
Ok..but now I'm directly fetching from `g_conf` and for `exporter_addr` default is set too.